lib/rules/indent.js

```diff
@@ -807,6 +889,71 @@ module.exports = {
             });
         }
 
+        /**
+        * Ignore first token of each line within an unknown node if its
+        * offset does not depend on another token's offset. If the first token
+        * of a line's offset depends on the offset of another token, make the
+        * last dependent offset depend on the first token of its line.
```

Comment:
Yes I will clean this up. Maybe If I create a separate function for handling lines with dependencies I can explain the logic in better detail.